### ISSUE TITLE
bug: LfMerge errors not making it to the UI

### ISSUE THREAD

**Issue by longrunningprocess** [sillsdev/web-languageforge#1303]
**Describe the bug**
I was recently working with a client who received a generic "Well this is embarrassing..." error during a S/R.  I did a little research and finally found the helpful message in the project's `.state` file that looks like it was intended for the end user.  I'm not sure where exactly the message got changed when being returned to the user.

```
Putting project 'project_name' on hold due to unhandled exception: \nChorus.VcsDrivers.Mercurial.UnrelatedRepositoryErrorException: The repository(a.k.a 'project' or 'collection') that you tried to synchronize with has the same name as yours, but it does not have the same heritage, so it cannot be synchronized. In order to Send/Receive projects, you have to start with a single project/collection, then copy that around. Don't feel bad if this is confusing, just ask for some technical help.
```

**To Reproduce**
The real problem was related to some versioning mismatch between the project on LF and the project on Language Depot.

**Expected behavior**
I would expect the user to receive the message about the "heritage"

**Additional context**
My research was on a project that is considered sensitive so additional details of my research are captured in email, just ping me for them.

**Comment by megahirt** [comment]
C.f. https://github.com/sillsdev/LfMerge/issues/183

**Comment by megahirt** [comment]
Yes, this would be a good improvement...

**Comment by rmunn** [comment]
Some (many?) LfMerge errors are not written with the end user in mind, though. But it might be possible to add some code to LfMerge, and/or the LF code that interacts with LfMerge, to filter error messages and pass through a specific list of error messages that *are* intended for the end user to see.

**Comment by megahirt** [comment]
This is actually required for the latest LFMerge changes that are being merged into develop.

**Comment by megahirt** [comment]
I anticipate that this issue will be fixed once #1444 is implemented.
